### PR TITLE
Introduce `KafkaListener.threadsValue` to allow for dynamic config

### LINKED ISSUE #689: Is there any way to set KafkaListener.threads() using config?
### Issue description

Hello,

Today I'm setting my consumer threads using KafkaListerner Annotation:

```java
@KafkaListener(groupId = "${audit-log.group-id}", offsetReset = OffsetReset.EARLIEST, threads =  3)
```

I'd like to put threads in my config (as I do with groupId) but thread is `int` so I can't use placeholder there.

Any recommendation in how to do it?

Thank you

### PR DESCRIPTION
**Pull request by guillermocalvo**
Closes #689

### CODE REVIEW

**Review by jeremyg484 — APPROVED** [review]
Looks good, though it is a breaking change as noted.

If we don't want to force another major version bump for something this simple, an alternate approach would be to keep the existing attribute as an `int` and add an alternate attribute to the annotation that is a `String`.

@sdelamo What do you think?

**Review by sdelamo — DISMISSED** [review]
we cannot do a breaking change now. Too late. either we search for a workaround or we do it for the next major version of Micronaut Kafka

**Comment by dstepanov** [comment]
I would keep it as an int and introduce `threadsValue`

File: kafka/src/test/groovy/io/micronaut/configuration/kafka/annotation/KafkaListenerSpec.groovy [hunk 1]
```diff
@@ -293,7 +293,7 @@ class KafkaListenerSpec extends AbstractEmbeddedServerSpec {
     }
 
     @Requires(property = 'spec.name', value = 'KafkaListenerSpec')
-    @KafkaListener(clientId = "kafka-consumer-with-configurable-number-of-threads", threads = '${my.thread.count}')
+    @KafkaListener(clientId = "kafka-consumer-with-configurable-number-of-threads", threadsValue = '${my.thread.count}')
```
Comment (jeremyg484) [inline kafka/src/test/groovy/io/micronaut/configuration/kafka/annotation/KafkaListenerSpec.groovy:296]: I'd like to see some additional test cases to also exercise setting the `threads` attribute, and then setting both `threadsValue` and `threads` at the same time.

**Review by jeremyg484 — CHANGES_REQUESTED** [review]

**Review by jeremyg484 — APPROVED** [review]

**Review by dstepanov — APPROVED** [review]

**Review by sdelamo — APPROVED** [review]

**Pull request merged**